### DISCUSSION TITLE
Unmatching d_germline_alignment indexes to the length of the d segment in BCR sequences

### DISCUSSION THREAD

**Discussion by Bar-Avidov** [milaboratory/mixcr#1532] [Q&A] [answered]
I used the exportAIRR command to annotate BCR sequences obtained from different tissues of a mouse.
In the output tsv file, I looked on the d_germline_start / d_germline_end columns and noticed that the indexes go beyond the range of indexes of the d segment annotation for the sequences, meaning the d segment is shorted than the end (and sometimes also the start) index in the table.

I have uploaded a screenshot of the some reads. for example in this case, the sequence was annotated with IDHD5-8  d segment which has a length of 10 nuceltodies: "GAGTACCTAC", however in the output of MIXCR the indexes are d_germline_start: 31 and d_germline_end: 39. how can the indexes be larger than the length of the d segment ?

<img width="1051" alt="Screenshot 2024-02-08 at 17 17 28" src="https://github.com/milaboratory/mixcr/assets/117773913/229c84ef-4e03-44f9-9a34-6636d5f8a7bd">

**Comment by mizraelson** [comment, marked as answer]
Hi,
MiXCR incorporates P segment detection, which means that the reference sequence for each gene includes parts accounting for the P segment. Specifically, for the D gene, this involves an inverted sequence both before and after the actual D gene to accommodate potential P nucleotides. For instance, the [IGHD5-8 gene is 29 nucleotides long](https://vdj.online/library/gene/10090:IGHD5-8). In the reference, the first 29 nucleotides represent the P segment before the D gene, followed by 29 nucleotides of the D gene itself, and then another 29 nucleotides for the P segment after the D gene. Therefore, a position marked as '31' indicates the 2nd nucleotide of the D gene. Conversely, a position marked as '24' would imply the presence of 6 P-nucleotides before the start of the D gene.

**Reply by ayeletperes** [reply]
Hey,

Does this apply also for the V germline end, the J germline start and end? 

sequence_id                 read.5
productive                       F
v_call                 IGHV3-66*01
d_call                 IGHD1-26*01
j_call                    IGHJ6*02
v_germline_start               1.0
v_sequence_start               1.0
v_germline_end               294.0
v_sequence_end               294.0
d_germline_start              13.0
d_sequence_start             295.0
d_germline_end                21.0
d_sequence_end               303.0
j_germline_start              28.0
j_sequence_start             310.0
j_germline_end                83.0
j_sequence_end               365.0

**Reply by mizraelson** [reply]
yes, V end and J start follow the same logic due to the P segment reference

**Comment by Bar-Avidov** [comment]
thanks for the fast reply,

I am trying to use the output data of MIXCR to add a
sequence_germline_d_mask column to the data using another tool (dowser
library in R).
I understand that the indexing of the d segment is not the conventional of
the AIRR format and this leads of problem when using other tools.

Is there an easy way to convert the indexing of the d segment that includes
the identification of the palindrome sequence, into indexing of only the d
segment ?
Is the palindrome sequence always the same length of the d segment ? In
such a case I can simply reduce from the start / end index the length of
the d segment to the indexing of the d segment?

thanks a head for the help,

bar avidov

**Reply by mizraelson** [reply]
Yes, the palindrome is always the same length as the D gene itself.

**Comment by Bar-Avidov** [comment]
Is there a way to convert the indexes into the indexes often used in the
AIRR format which include only the indexes of the d segment germ line part
without the palinedrom ?

**Reply by mizraelson** [reply]
Currently we do not have such conversion, but will add it in the future.
